src/openms/source/FORMAT/OMSFileLoad.cpp

```diff
@@ -1130,4 +1134,237 @@ namespace OpenMS::Internal
+      json_data.insert(table, exportQueryToJSON_(sql, {"id", "parent_id", "ps_id", "processing_step_id", "st_id",
+                                                       "score_type_id", join_key}));
+    }
+
+    exportMetaInfoToJSON_(json_data, parent_table, join, order, join_key);
+  }
+
+
+  void OMSFileLoad::exportToJSON(ostream& output)
+  {
+    // @TODO: this constructs the whole JSON file in memory - write directly to stream instead?
+    // (more code, but would use less memory)
+    QJsonObject json_data;
+    QString sql;
+    // version:
+    if (tableExists_(db_name_, "version"))
+    {
+      sql = "SELECT * FROM version";
+      json_data.insert("version", exportQueryToJSON_(sql));
+    }
+    // input files:
+    if (tableExists_(db_name_, "ID_InputFile"))
+    {
+      sql = "SELECT * FROM ID_InputFile ORDER BY name";
+      json_data.insert("ID_InputFile", exportQueryToJSON_(sql));
+    }
+    // score types:
+    // this query is used in multiple joins below (@TODO: replace with DB view?):
+    subquery_score_ = "SELECT ID_ScoreType.id AS st_id, accession AS score_type_accession, name AS score_type_name FROM ID_ScoreType JOIN CVTerm ON cv_term_id = CVTerm.id";
+    if (tableExists_(db_name_, "ID_ScoreType") && tableExists_(db_name_, "CVTerm"))
+    {
+      sql = "SELECT * FROM ID_ScoreType JOIN CVTerm ON cv_term_id = CVTerm.id ORDER BY accession, name";
+      json_data.insert("ID_ScoreType", exportQueryToJSON_(sql, {"id", "cv_term_id"}));
+    }
+    // processing software:
+    if (tableExists_(db_name_, "ID_ProcessingSoftware"))
+    {
+      sql = "SELECT * FROM ID_ProcessingSoftware ORDER BY name, version";
+      json_data.insert("ID_ProcessingSoftware", exportQueryToJSON_(sql));
+      // processing software - assigned scores:
+      if (tableExists_(db_name_, "ID_ProcessingSoftware_AssignedScore"))
+      {
+        // need to replace DB keys for both software and score type with corresponding values:
+        QString subquery_software = "SELECT id AS sw_id, name AS software_name, version AS software_version FROM ID_ProcessingSoftware";
+        sql = "SELECT software_name, software_version, score_type_accession, score_type_name, score_type_order FROM ID_ProcessingSoftware_AssignedScore JOIN (" + subquery_software + ") ON software_id = sw_id JOIN (" + subquery_score_ + ") ON score_type_id = st_id ORDER BY software_name, software_version, score_type_order";
+        json_data.insert("ID_ProcessingSoftware_AssignedScore", exportQueryToJSON_(sql, {}));
+      }
+    }
+    // processing steps:
+    if (tableExists_(db_name_, "ID_ProcessingStep"))
+    {
+      // processing steps potentially get referenced a lot, but their "identity" is complex
+      // (software + input files + timestamp) - create a view that includes an index number for referencing:
+      // @TODO: include input files in ordering (via "GROUP_CONCAT")
```

Comment:
how is the order determined right now?

Reply:
Just software name, version, processing date/time. See the `ORDER BY ...` in the next line.